Fix for "Cave Revamp" update

developer9998 opened the pull request:
- Fix object paths for computer locations and scene renderers
- Added more error notices during initalizing for easier bug fixing
- Made StartZone not be set right away to prevent computers from not spawning correctly
- Fix PhotonNetworkController references in BaseGameInterface
- Other minimal changes when initalizing CustomScreenInfo (Parenting done so you can't see monitors through other maps)

developer9998 committed:
Fix for update

- Fix object paths for computer locations and scene renderers
- Added more error notices during initalizing for easier bug fixing
- Made StartZone not be set right away to prevent computers from not spawning correctly
- Fix PhotonNetworkController references in BaseGameInterface
- Other minimal changes when initalizing CustomScreenInfo (Parenting done so you can't see monitors through other maps)

developer9998 commented:
- Disable the Reflection Probe built into the computer asset as they're now being used in the base game and they were causing objects near computers to display weirdly due to the Probe's settings

brodog1000 commented:
can it release yet?

developer9998 commented:
no

brodog1000 commented:
Why?

ghost commented:
> Why?

The Person who made computer interface Has to merge it.

developer9998 committed:
Update CustomComputer.cs

This is mainly done to fix the execution time for disabling mods disabled by CI, without these changes on initialized all mods would be visible no matter if they were disabled or not

developer9998 commented:
For those trying to "fix" these mods:
- These fixes aren't verified, and could cause game breaking issues
- Please don't spread your "methods" and sketchy files
- Please don't download any sketchy files which should "fix the mod"

I recommend you to just wait until a verified fix gets released for each mod, including this one.

developer9998 committed:
Improve initialization process

developer9998 commented:
Phew that's been bugging me for a bit

ghost commented:
Hopefully this gets merged

developer9998 committed:
General fixes

- Applied zone data for finding the combinedScene object
- Fixed non-treehouse keys not loading due to some of them lacking a material

developer9998 committed:
Fix AssetBundle

- Switch back to older model, most people think it's way better anyway
- Switched some references of "IComputerView" to "ComputerView" because I thought it would fix and error but turns out the error was caused by something else but whatever
- Updated position and such for monitor
- Fixed removing monitor for Stump

developer9998 committed:
Temporarily disable installation of the CommandRegistrar

developer9998 commented:
TODO: 
- Fix installation of the CommandRegistrar as for me atleast a Zenject error would occur

developer9998 committed:
Monitor update

Minor Tweaks:
- Activated commands and cleaned up the CommandRegistrar class
- Updated setting voice chat state and instrument volume
- For commands pressing the Up arrow will return your previous command after execution, based off of the custom computer that was on Quest, feature shown off here: https://youtu.be/DoUaos80V_M?si=nKWxfddkME_9fp66&t=455
- Renamed Computer Display Settings to Computer Settings
- Adjusted Group join info
- Adjusted Item settings info

Major Tweaks:
- Added different monitors
- There are two monitors, one slim "Classic" and another bulky "Modern"
- Monitors can be changed at any time in the Computer Settings tab or using BepInEx configuration
-  Rebuilt AssetBundle for the mod to include both monitors, more optimizations, and fix for shaders since the game uses URP now

developer9998 commented:
For those trying to "fix" these mods:
- These fixes aren't verified, and could cause game breaking issues
- Please don't spread your "methods" and sketchy files
- Please don't download any sketchy files which should "fix the mod"

I recommend you to just wait until a verified fix gets released for each mod, including this one.

developer9998 committed:
HOTFIX: Fix voice chat refresh

Fall3n5lay3r commented:
So is computer interface been fixed?

ghost commented:
> So is computer interface been fixed?
> 

Use at your own risk

Fall3n5lay3r commented:
great

Fall3n5lay3r commented:
btw i just tested it and there is still a couple things wrong with it. the normal computer which is normally behind it is not there anymore.when you go on to a mod channel like "monke map loader" it freezes entirely.

developer9998 commented:
The "normal computer" is supposed to go away, I also fixed that. MonkeMapLoader is currently broken, so wait for the mod to be fixed.

developer9998 commented:
Check my comment > btw i just tested it and there is still a couple things wrong with it. the normal computer which is normally behind it is not there anymore.when you go on to a mod channel like "monke map loader" it freezes entirely.

Fall3n5lay3r commented:
ok

Fall3n5lay3r commented:
the freezing thing happens on all mod option pages.Is that normal?

developer9998 commented:
freezing? please be more specific

Fall3n5lay3r commented:
i will dm you a vid of what happens
my user is Fall3nVR

developer9998 committed:
Polish

developer9998 commented:
- Improved commands (Outputs, and ComputerViews)

<img src="https://github.com/ToniMacaroni/ComputerInterface/assets/81720436/1b24fa2a-48f9-4651-bec2-dfb83463adba" width=70% height=70%><br>

- Updated monitor location names to better fit their location
- Made the monitor handling more simple and optimized
- Updated details view

<img src="https://github.com/ToniMacaroni/ComputerInterface/assets/81720436/10374014-25d9-4fc2-baa9-67c02656994b" width=50% height=50%><br>

- Other minor tweaks

developer9998 committed:
Mod fixes

developer9998 committed:
Hotfix !!

developer9998 commented:
"Mod fixes"
- Fixed custom views not ready for the update not displaying any text
- Updated ModList and Mod views to be better organized and provide more information (e.g. instructions, toggleable mods, viiew names)
<img src="https://github.com/ToniMacaroni/ComputerInterface/assets/81720436/7b7a1208-c2ac-4163-bf43-b646c4482d4a" width=60% height=50%>

"Hotfix !!"
- Fixed issue with mod page where text displaying "`This mod doesn't support toggling between Enabled/Disabled states.`" would be a dark green instead of a dark red, whoops

developer9998 committed:
Minor tweak

- Added a warning view to display whenever the game is outdated or the system isn't connected to internet
- Other things, I dunno

developer9998 commented:
Last commit most definitely needs some polish

Graicc commented on ComputerInterface/CustomKeyboardKey.cs:
```diff
@@ -87,9 +87,28 @@ public void Init(CustomComputer computer, EKeyboardKey key, Text keyboardText, s
         {
             Init(computer, key, keyboardText, text);
+
+            if (_material == null)
+            {
+                _originalColor = buttonColor;
+
+                var baseRenderer = GetComponent<Renderer>();
+                if (baseRenderer.material == null)
+                {
+                    _material = new Material(Shader.Find("Legacy Shaders/Diffuse"))
+                    {
+                        color = buttonColor
+                    };
+                    goto PrepareColour;
+                }
+                baseRenderer.material.color = buttonColor;
+                goto PrepareColour;
+            }
+
             _material.color = buttonColor;
             _originalColor = buttonColor;
 
+            PrepareColour:
```
This logic doesn't make sense, why is `_originalColor` always set to the new color? Also, GOTO considered harmfull, you should almost never use it

Graicc committed:
First pass minor edits

Graicc commented on ComputerInterface/Views/WarnView.cs:
```diff
@@ -0,0 +1,91 @@
+            }
+
+            public String getWarningMessage() => message;
+        }
+
+        public class OutdatedWarning : IWarning
+        {
+            public String getWarningMessage() => "You aren't on the latest version of Gorilla Tag, please update your game to continue playing with others.";
+        }
+
+        public class NoInternetWarning : IWarning
+        {
+            public String getWarningMessage() => "You aren't connected to an internet connection, please connect to a valid connection to continue playing with others.";
+        }
+
+        public class TemporaryBanWarning : IWarning
+        {
+            private String reason;
+            private int hoursRemaining;
+
+            public TemporaryBanWarning(String reason, int hoursRemaining)
+            {
+                this.reason = reason;
+                this.hoursRemaining = hoursRemaining;
+            }
+
+            public String getWarningMessage() => $"You have been temporarily banned. You will not be able to play with others until the ban expires.\nReason: {reason}\nHours remaining: {hoursRemaining}";
+        }
+
+        public class PermanentBanWarning : IWarning
+        {
+            private String reason;
+
+            public PermanentBanWarning(String reason)
+            {
+                this.reason = reason;
+            }
+
+            public String getWarningMessage() => $"You have been permanently banned.\nReason: {reason}";
+        }
```
Doing this [makes invalid states unrepresentable](https://fsharpforfunandprofit.com/posts/designing-with-types-making-illegal-states-unrepresentable/). We never have to worry about object[] warnParams having the wrong number of params, or params of the wrong types, and every warning will always have a message associated with it (if there was a new warning type, it would be easy to forget to update the switch to handle it)

gizmogoat commented:
should we provide the source unity project for the monitor assets somewhere? maybe inside the resources folder alongside the assetbundle? it could make contributions easier in the future if others don't have to rip the resources and potentially break them along the way

developer9998 committed:
Monitor updates

developer9998 committed:
Update ComputerSettingsView.cs

SawyerISnotepic commented:
how do i install this?

ghost commented:
Does this fix still work?

ghost commented:
It doesn't load in Forest

gizmogoat commented:
the zone indexes changed recently so it only works in certain maps, its very easy to fix but it looks like that hasnt been done to this PR yet

ghost commented:
Ah okay

CrytosGtag commented:
Btw dev i made a Pull Request on your Computer Interface that fixes where the computer spawns
